+bids/layout.m

```diff
@@ -365,36 +378,56 @@
   pth = fullfile(subject.path, modality);
   fullpath_filename = fullfile(pth, file);
 
-  subject.(modality)(end).metafile = bids.internal.get_meta_list(fullpath_filename);
-  subject.(modality)(end).dependencies.explicit = {};
-  subject.(modality)(end).dependencies.data = {};
-  subject.(modality)(end).dependencies.group = {};
+  if ~isfield(subject.(modality)(end), 'dependencies') || ...
+          isempty(subject.(modality)(end).dependencies)
+    subject.(modality)(end).dependencies.explicit = {};
+    subject.(modality)(end).dependencies.data = {};
+    subject.(modality)(end).dependencies.group = {};
+  end
 
-  ext = subject.(modality)(end).ext;
-  suffix = subject.(modality)(end).suffix;
-  pattern = strrep(file, ['_' suffix ext], '_[a-zA-Z0-9.]+$');
-  candidates = bids.internal.file_utils('List', pth, ['^' pattern '$']);
-  candidates = cellstr(candidates);
+  if strncmp(dep_prev_files.data_base, file, dep_prev_files.data_len)
 
-  for ii = 1:numel(candidates)
+    %     subject.(modality)(end + 1, 1) = subject.(modality)(end, 1);
+    %     subject.(modality)(end, 1).ext = file(dep_prev_files.data_len:end);
+    %     subject.(modality)(end, 1).filename = file;
 
-    if strcmp(candidates{ii}, file)
-      continue
-    end
+    dep_fname = fullfile(pth, subject.(modality)(end - 1, 1).filename);
+    subject.(modality)(end).dependencies.data{end + 1} = dep_fname;
```

Comment:
This will work only if last subject is clone of subject before it -- see first comment.

Overwise a for loop is needed starting from `index_data` to `end - 1`

Reply:
Same remark for metafile field. If last file is not clone, you need to copy the metafile attribute.